src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -428,83 +450,74 @@ private async Task RefreshIndividualKeyValues(CancellationToken cancellationToke
+                        ProcessChanges(Enumerable.Repeat(keyValueChange, 1));
+                    }
+                }
+                else
+                {
+                    // Load the key-value in case the previous load attempts had failed
+                    var options = new SettingSelector { LabelFilter = watchedLabel };
+
+                    try
+                    {
+                        await CallWithRequestTracing(async () => watchedKv = await client.GetConfigurationSettingAsync(watchedKey, watchedLabel, cancellationToken).ConfigureAwait(false)).ConfigureAwait(false);
+                    }
+                    catch (RequestFailedException e) when (e.Status == (int)HttpStatusCode.NotFound)
+                    {
+                        watchedKv = null;
+                    }
+
+                    UpdateCacheExpirationTime(changeWatcher, success: true);
+
+                    if (watchedKv != null)
+                    {
+                        if (changeWatcher.RefreshAll)
                         {
-                            watchedKv = null;
+                            shouldRefreshAll = true;
+                            break;
                         }
 
-                        success = true;
+                        hasChanged = true;
 
-                        if (watchedKv != null)
+                        // Add the key-value if it is not loaded, or update it if it was loaded with a different label
+                        _applicationSettings[watchedKey] = watchedKv;
+                        _watchedSettings[watchedKeyLabel] = watchedKv;
+
+                        // Invalidate the cached Key Vault secret (if any) for this ConfigurationSetting
+                        foreach (IKeyValueAdapter adapter in _options.Adapters)
                         {
-                            if (changeWatcher.RefreshAll)
-                            {
-                                shouldRefreshAll = true;
-                                break;
-                            }
-
-                            hasChanged = true;
-
-                            // Add the key-value if it is not loaded, or update it if it was loaded with a different label
-                            _applicationSettings[watchedKey] = watchedKv;
-                            _watchedSettings[watchedKeyLabel] = watchedKv;
-
-                            // Invalidate the cached Key Vault secret (if any) for this ConfigurationSetting
-                            foreach (IKeyValueAdapter adapter in _options.Adapters)
-                            {
-                                adapter.InvalidateCache(watchedKv);
-                            }
+                            adapter.InvalidateCache(watchedKv);
                         }
                     }
                 }
-                finally
-                {
-                    UpdateCacheExpirationTime(changeWatcher, success);
-                }

```

Comment:
I think we should still keep `UpdateCacheExpirationTime` in the `finally` clause even if you just want to do it for the success case. Your current code missed out cache expiration update in line 459.

Reply:
The comment is not relevant anymore.